### PR TITLE
enable Javascript in iOS, support abort loading specific URLs

### PR DESCRIPTION
**Pull request by hnvn**
I create this pull request to give WebView few features:
- enable Javascript in iOS.
- support abort loading specific URLs by using regex pattern. Native webview in Android and iOS will use this regex to determine to continue or abort loading an URL in method `shouldOverrideUrlLoading` in Android and `webView:decidePolicyForNavigationAction:decisionHandler` in iOS

### CODE REVIEW

**Comment by sameerbakre** [comment]
Looks to be perfect answer to most of the problem we are having with this plugin.
When can we expect this to be available for downloading?

File: example/lib/main.dart [hunk 1]
```diff
@@ -201,6 +201,7 @@ class _MyHomePageState extends State<MyHomePage> {
                   selectedUrl,
                   rect: Rect.fromLTWH(0.0, 0.0, MediaQuery.of(context).size.width, 300.0),
                   userAgent: kAndroidUserAgent,
+                  invalidUrlRegex: r'^(https).+(twitter)', // prevent redirecting to twitter when user click on its icon in flutter website
```
Comment (charafau) [inline example/lib/main.dart]: This code doesn't work for me on iOS simulator

Comment (hnvn) [reply example/lib/main.dart]: I still get it worked in my iOS simulator. Note that: `invalidUrlRegex` is only set up for WebView launched by 'Open Webview (rect)' button.

Comment (charafau) [reply example/lib/main.dart]: Ah yes, you're right. It works now. Thank you!

**Comment by evelyne24** [comment]
> enabling Javascript in iOS works as expected but for some reason it doesn't block URL redirection - at least when I click on twitter logo on flutter website, it gets redirected. Tested on iOS simulator

The `invalidUrlRegex` doesn't appear to work with the `WebviewScaffold`?

**Comment by hnvn** [comment]
Have a look at: https://github.com/fluttercommunity/flutter_webview_plugin/pull/292#discussion_r269172821

**Comment by Raggaer** [comment]
How should I stop the loading tho? using `invalidUrlRegex` on the Scaffold works fine (blocks the redirect) but I want to detect the url and fire my own event, is that possible?

`onUrlChanged` only fires when the website is fully loaded.

Thanks

**Comment by yorickreum** [comment]
@Raggaer Did you solve this problem? Facing the same issue…

**Comment by Raggaer** [comment]
@yorickreum I could not solve this sadly. Hopefully someone has some information to share with us

**Comment by yorickreum** [comment]
My approach today was to listen to the state change (load abort event). This kind of works, it only gets fired twice (not once), that's quite annoying...

**Comment by yorickreum** [comment]
Still no solution...

**Comment by sameerbakre** [comment]
@yorickreum , yes, please use onStateChanged method available, you can use state.type == WebViewState.shouldStart to listen it for once only.

**Pull request merged**